### ISSUE TITLE
Area selection cursor is shown in final screenshot

### ISSUE THREAD

**Issue by diniamo** [hyprwm/contrib#60]
![image](https://github.com/hyprwm/contrib/assets/55629891/afb3789a-9770-43be-8667-27f511d3682b)

As you can see in the screenshot, the cursor I selected the area with is shown. I didn't pass `--cursor`.

**Comment by fufexan** [comment]
Can you post the full command you've used to take the screenshot? Also, are you using NVidia/have the `WLR_NO_HARDWARE_CURSORS=1` variable set?

**Comment by diniamo** [comment]
Yes, I use nvidia, no I don't have it set, but it doesn't seem to make a difference if I do.
```
bind = $ctrl, Print, exec, ~/.local/bin/grimblast --notify --freeze copy area
```
And I also made a small modification in grimblast where slurp is called:
```sh
slurp -c ed8796 -b 00000000 -s 00000000 -B 00000000
```

**Comment by fufexan** [comment]
Sounds like a nvidia issue, as I don't see the cursor when running the same command.
![image](https://github.com/hyprwm/contrib/assets/36706276/826a4985-8219-4736-9f85-fa383bac7f4e)
As can be seen, my hyprpicker is built from a commit in `2023-10-05`, so check if yours might be out of date.

**Comment by diniamo** [comment]
I have the latest/same commit.

**Comment by kRHYME7** [comment]
We found out that grimblast captures hyprpicker cursor. Should we request a feature on hyprpicker  to make cursor transparent? Or should we implement a separate package for freezing?

**Comment by fufexan** [comment]
I haven't seen that, mind showing an example? Maybe record a video of it happening?

**Comment by kRHYME7** [comment]
Sorry for the very late response. Here is my env also if this could cause something.
```
❯ cat ~/.config/hypr/userprefs.conf | grep 'env'
env = GDK_SCALE,1
env = WLR_NO_HARDWARE_CURSORS,1
env = WLR_EGL_NO_MODIFIERS,1
env = XDG_CURRENT_DESKTOP,Hyprland
env = XDG_SESSION_TYPE,wayland
env = XDG_SESSION_DESKTOP,Hyprland
env = GDK_BACKEND,wayland,x11
env = QT_QPA_PLATFORM,wayland
env = SDL_VIDEODRIVER,wayland
env = CLUTTER_BACKEND,wayland
env = QT_AUTO_SCREEN_SCALE_FACTOR,2
env = QT_WAYLAND_DISABLE_WINDOWDECORATION,1
❯ cat ~/.config/hypr/hyprland.conf | grep 'env'
exec-once = dbus-update-activation-environment --systemd WAYLAND_DISPLAY XDG_CURRENT_DESKTOP # for XDPH
exec-once = dbus-update-activation-environment --systemd --all # for XDPH
exec-once = systemctl --user import-environment WAYLAND_DISPLAY XDG_CURRENT_DESKTOP # for XDPH
# Some default env vars.
env = XDG_CURRENT_DESKTOP,Hyprland
env = XDG_SESSION_TYPE,wayland
env = XDG_SESSION_DESKTOP,Hyprland
env = GDK_BACKEND,wayland
env = QT_QPA_PLATFORM,wayland
env = QT_QPA_PLATFORMTHEME,qt5ct
env = QT_WAYLAND_DISABLE_WINDOWDECORATION,1
env = QT_AUTO_SCREEN_SCALE_FACTOR,1
env = MOZ_ENABLE_WAYLAND,1
```

![samp](https://github.com/hyprwm/contrib/assets/53417443/85468dc9-b713-407f-80c2-1447e58d7a5f)
![samp2](https://github.com/hyprwm/contrib/assets/53417443/75c76d14-7e42-46dd-b410-74a0896e8790)

https://github.com/hyprwm/contrib/assets/53417443/3836226c-7273-450b-a965-ce52273926f7

Thanks :pray:

**Comment by fufexan** [comment]
My guess is `env = WLR_NO_HARDWARE_CURSORS,1` is causing the cursor to keep showing.

**Comment by diniamo** [comment]
Even in non-nvidia environments, there are issues without that variable, but this should be fixable in the screenshot stack's code, right?

**Comment by fufexan** [comment]
Not sure what "screenshot stack" refers to here, but it's likely a bug with either slurp or hyprpicker.

**Comment by diniamo** [comment]
It refers to grim, slurp, and hyprpicker.

**Comment by vaxerski** [comment]
none of these are the culprit, it's hyprland's fault. FWIW, same will happen on any wlroots compositor. We could get around it, but it would be inefficient.

**Comment by waddyhussain** [comment]
Had this issue for a while, but only just found out about the workaround of removing --freeze (or uninstalling hyprpicker if you don't use it). Had to look through one of the other threads to find it, so posting here in case anyone comes across this issue while searching.

**Comment by diniamo** [comment]
Removing freeze makes it even worse for me, as now both the cursor and the box I used to select the area are present in the screenshot.

**Comment by Ilirski** [comment]
Not sure if it's related, but I have to double click in order to finish selecting area when doing`env GRIMBLAST_EDITOR=gimp grimblast --notify --freeze edit area` 

https://github.com/hyprwm/contrib/assets/24876805/fdba20e1-4b65-4b7e-adcb-a89ea0a39a6d

Removing `--freeze` makes it open editor on click as expected.

**Comment by PacificViking** [comment]
I am having the same error with WLR_NO_HARDWARE_CURSORS=1 on Hyprland/Nix with both grim and grimblast.

**Comment by carschandler** [comment]
Also have this error on Hyprland/NixOS/nVidia with `WLR_NO_HARDWARE_CURSORS=1`

**Comment by leon-erd** [comment]
For me omitting the `--freeze` option solves the problem of the cursor being in the image (but I'm also not using nvidia 
/`WLR_NO_HARDWARE_CURSORS=1`).
Regarding the area selection box being visible in the image: I realized that the problem was due to the screenshot being taken before the closing layer animation of the selection was finished.
Disabling the animations for the selection in hyprland with:
```
layerrule=noanim, hyprpicker
layerrule=noanim, selection
```
Solved that for me! Maybe it helps someone :)

**Comment by diniamo** [comment]
WLR_NO_HARDWARE_CURSORS=1 should be removed by everyone at this point, it's been fixed in git for a while, and the latest stable release has the patch too.

That also means we can solve this issue I guess.

**Comment by roastedcheese** [comment]
getting this again on -git and without `WLR_NO_HARDWARE_CURSORS=1`

**Comment by diniamo** [comment]
Same, I forgot to report

**Comment by vaxerski** [comment]
git ignores the env, it uses cursor:no_hardware_cursors instead

**Comment by roastedcheese** [comment]
don't have that set either

**Comment by diniamo** [comment]
For anyone with this issue, `cursor:allow_dumb_copy = true` seems to fix it after the aq rewrite. I don't know if it has any downsides, I haven't noticed the hitch the hl wiki mentions.

**Comment by kRHYME7** [comment]
Is this device dependent? 

# no_hardware_cursors = true

![image](https://github.com/user-attachments/assets/b58ce60a-1bb1-4d60-85f0-442b888b3c5b)
![image](https://github.com/user-attachments/assets/9ca5c374-3e90-4c7c-9b15-36f19bf4d8c6)
![image](https://github.com/user-attachments/assets/de50548d-8d6e-4dbf-aea9-002468b5592b)
![image](https://github.com/user-attachments/assets/8a1ff15c-6899-4104-a096-adc43480c233)

# no_hardware_cursors = false
![image](https://github.com/user-attachments/assets/95007bb9-05f2-493d-99db-bc14e00020f0)
![image](https://github.com/user-attachments/assets/c9fa25f6-4e0a-4490-8a0b-abf67516ec5a)

**Comment by diniamo** [comment]
You should have `no_hardware_cursors = false`

**Comment by different-name** [comment]
I'm getting this bug with `no_hardware_cursors = false`

**Comment by m2neral** [comment]
![2024-10-16T22:28:19,049685288+03:00](https://github.com/user-attachments/assets/7c4530a5-b839-443f-ad29-12bcaa920edb)
![2024-10-16T22:29:07,164581156+03:00](https://github.com/user-attachments/assets/90103cc5-74a9-47bb-a097-a55be9af7b48)

I tried everything on this page and it still doesn't work with or without --freeze. I'm on nvidia.

**Comment by LeonardoMor** [comment]
[Relevant](https://todo.sr.ht/~emersion/grim/93?__goaway_challenge=meta-refresh&__goaway_id=c58b0a6ec74c9a4c5ac61c923c3881bb&__goaway_referer=https%3A%2F%2Fforum.hypr.land%2F)
